crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -1968,3 +1993,40 @@ pub async fn create_test_mqtt_client_with_empty_operations() -> Connection {
     let mapper_config = create_mapper_config(&Operations::default());
     create_test_mqtt_client(&mapper_config).await
 }
+
+pub async fn create_c8y_mapper(
+    app_name: &str,
+    mqtt_host: String,
+    mqtt_port: u16,
+    converter: Box<dyn Converter<Error = ConversionError>>,
+) -> Result<Mapper, anyhow::Error> {
+    let health_check_topics: TopicFilter = health_check_topics(app_name);
+
+    let mapper_config = converter.get_mapper_config();
+    let mut topic_filter = mapper_config.in_topic_filter.clone();
+    topic_filter.add_all(health_check_topics.clone());
+
+    let mqtt_client =
+        Connection::new(&mqtt_config(app_name, &mqtt_host, mqtt_port, topic_filter)?).await?;
+
+    let mqtt_config = mqtt_channel::Config::default()
+        .with_host(mqtt_host)
+        .with_port(mqtt_port)
+        .with_last_will_message(service_monitor_status_message(
+            "test-device",
+            app_name,
+            "down",
+            "service",
+            None,
+        ));
+
+    let _lwp_mqtt_client = Connection::new(&mqtt_config).await?;
```

Comment:
This doesn't feel right. What we wanted to test was the mapper's contract that creates the dedicated MQTT client for last will message and see if that logic works. But here, instead of testing that logic of the actual mapper, we're creating a test client mimicking the same behaviour and testing that. That's like unit testing a test entity. I'm sure that's not what we wanted.

Reply:
I was trying to mimic the tedge-mapper. It does not make sense here in this case. It's easy to implement an `integration` test. So, I removed this test.